packages/cli/src/ui/components/messages/ToolMessage.tsx

```diff
@@ -68,11 +71,61 @@ export const ToolMessage: React.FC<ToolMessageProps> = ({
   const { renderMarkdown } = useUIState();
   const isAlternateBuffer = useAlternateBuffer();
   const isThisShellFocused =
-    (name === SHELL_COMMAND_NAME || name === 'Shell') &&
+    (name === SHELL_COMMAND_NAME ||
+      name === 'Shell' ||
+      name === SHELL_TOOL_NAME) &&
     status === ToolCallStatus.Executing &&
     ptyId === activeShellPtyId &&
     embeddedShellFocused;
 
+  const { setEmbeddedShellFocused } = useUIActions();
+  const containerRef = React.useRef<DOMElement>(null);
+  // The shell is focusable if it's the shell command, it's executing, and the interactive shell is enabled.
+  const isThisShellFocusable =
+    (name === SHELL_COMMAND_NAME ||
+      name === 'Shell' ||
+      name === SHELL_TOOL_NAME) &&
+    status === ToolCallStatus.Executing &&
+    config?.getEnableInteractiveShell();
+
+  useMouse(
+    (event: MouseEvent) => {
+      if (isThisShellFocusable && event.name === 'left-press') {
```

Comment:
Core logic here is now duplicated with InputPrompt. Lets add a a helper to share the logic
maybe something like
useMouseClick(containerRef, (event, x, y) => {
});

that only gets called for clicks actually within the containerRef's bounding box.
Open to other apis for this as well, that one is half baked. Could also be a Focusable component.